cluster-api-gcp/deploy/deploy_helper.go

```diff
@@ -306,14 +315,19 @@ func (d *deployer) waitForServiceAccount(timeout time.Duration) error {
 	if err != nil {
 		return err
 	}
-	startTime := time.Now()
 
-	for time.Now().Sub(startTime) < timeout {
-		_, err := client.CoreV1().ServiceAccounts(ServiceAccountNs).Get(ServiceAccountName, metav1.GetOptions{})
+	eb := util.NewExponentialBackoff(timeout, 10 * time.Second)
+	backoff.Retry(func() error {
+		_, err = client.CoreV1().ServiceAccounts(ServiceAccountNs).Get(ServiceAccountName, metav1.GetOptions{})
 		if err == nil {
```

Comment:
this function call can be part of the if check, - 
if _, err = client......(); err == nil {
    return nil
}

Reply:
Since the line is pretty long as it is, I prefer it here. + consistency with the rest of the file.